### PR TITLE
🖥️ Create tables and execute first data migrations

### PR DESCRIPTION
**Pull request by FedericoDM**
## Describe your changes

- Adjusted `main.py` file to import data models for migration purposes.
- Created `database.py` file with models for our tables, along with their relationships. Note that I added several association tables for many-to-many relationships.
- Primary location is 'boolean' now. 
- Executed the migration process, which creates the `migrations/` directory automatically. The directory contains several relevant files, along with the `versions` folder.
- Adjusted `data_model.md` file with new association tables and minor adjustments.

NOTE: The `.py` scripts are automatically generated with each migration. You will see several files, which are all of the migrations that have been created.

## Non-obvious technical information

I ran a couple of commands to create the migration:

- First, navigate to the `new_arrivals_chi` folder.
- Then, run `flask --app app/main.py db migrate -m "Initial migration"`, where you can edit the migration name.
- The previous command will create a .py file with the SQL-ish commands for the migration (either creating tables, etc). Some adjustments may be needed.
- Then, run the ` flask --app app/main.py db upgrade` to see the changes on the database.

## Checklist before requesting a review
- [x] pre-commit run --all-files (run before pushing)

```commandline

new-arrivals-chi-py3.11) root@LAPTOP-RMMEN33V:/mnt/c/Users/fdmol/Desktop/MSCAPP/new-arrivals-chi/new_arrivals_chi/app# flask --app main.py db migrate -m "Initial migration"
INFO  [alembic.runtime.migration] Context impl PostgresqlImpl.
INFO  [alembic.runtime.migration] Will assume transactional DDL.
INFO  [alembic.autogenerate.compare] Detected added table 'hours'
INFO  [alembic.autogenerate.compare] Detected added table 'languages'
INFO  [alembic.autogenerate.compare] Detected added table 'locations'
INFO  [alembic.autogenerate.compare] Detected added table 'services'
INFO  [alembic.autogenerate.compare] Detected added table 'organizations'
INFO  [alembic.autogenerate.compare] Detected added table 'service_dates'
INFO  [alembic.autogenerate.compare] Detected added table 'languages_organizations'
INFO  [alembic.autogenerate.compare] Detected added table 'organizations_hours'
INFO  [alembic.autogenerate.compare] Detected added table 'organizations_services'
INFO  [alembic.autogenerate.compare] Detected added table 'users'

  Generating /mnt/c/Users/fdmol/Desktop/MSCAPP/new-arrivals-chi/new_arrivals_chi/app/migrations/versions/2024-04-25-2341_16cf46fcdfbf_initial_migration.py ...  done

(new-arrivals-chi-py3.11) root@LAPTOP-RMMEN33V:/mnt/c/Users/fdmol/Desktop/MSCAPP/new-arrivals-chi/new_arrivals_chi/app# flask --app main.py db upgrade
INFO  [alembic.runtime.migration] Context impl PostgresqlImpl.
INFO  [alembic.runtime.migration] Will assume transactional DDL.
INFO  [alembic.runtime.migration] Running upgrade  -> 16cf46fcdfbf, Initial migration

```

### CODE REVIEW

File: new_arrivals_chi/app/migrations/README
Comment (klinkoberstar) [inline new_arrivals_chi/app/migrations/README:1]: don't think it needs to be resolved in this PR but might want to add a bit more documentation to the readme and/or throughout just keeping in mind we may be handing over ownership down the road!

File: new_arrivals_chi/app/database.py
Comment (MadeleineKRoberts) [reply new_arrivals_chi/app/database.py:136]: also should there be a reference to location ? lmk your thoughts!

File: new_arrivals_chi/app/database.py
Comment (MadeleineKRoberts) [reply new_arrivals_chi/app/database.py:146]: should there be a FK to services  ? lmk your thoughts!

**Comment by FedericoDM** [comment]
@michplunkett Could you take a look at the `new_arrivals_chi/app/database.py` file and see if everything makes sense? Mainly about the relationships part (some of them are 'many to many', so I had to create some association tables).

File: new_arrivals_chi/app/migrations/versions/2024-04-25-2341_16cf46fcdfbf_initial_migration.py
Comment (MadeleineKRoberts) [inline new_arrivals_chi/app/migrations/versions/2024-04-25-2341_16cf46fcdfbf_initial_migration.py:20]: Would need to be adapted to comments in database.py -- sorry for the hassle!

**Review by MadeleineKRoberts — CHANGES_REQUESTED** [review]
Left in line comments about conversations I had with Michael at OH -- feel free to message me to chat about the questions and comments. Comparing to the models_m4 branch could be helpful to make sure we are on the same page!

**Review by klinkoberstar — CHANGES_REQUESTED** [review]
Left a few comments throughout!

The only thing that I want to make sure is resolved before merging is that we fix deleting/overwriting some of the things in the `docs/` folder! I know merging that has been a bit messy so let me know if I can help - hopefully it's as simple as syncing this branch with the` docs/` folder in main.

Long term we may want a bit more documentation throughout (as someone with a bit more of an outside perspective on this part of the project - and knowing we may pass the code base off at some point). By no means is this necessary for this PR tho!

INCREDIBLE work @FedericoDM and @MadeleineKRoberts  🥇

File: new_arrivals_chi/app/config.py
Comment (michplunkett) [inline new_arrivals_chi/app/config.py:7]: Since `DEBUG` is true for all but one category, you can default it to true here and just set it to false in `ProductionConfig`.

File: new_arrivals_chi/app/database.py
Comment (michplunkett) [reply new_arrivals_chi/app/database.py:51]: If you only see two roles, you can have a boolean field that `is_admin` or you can do a field that only accepts certain values.

File: new_arrivals_chi/app/database.py
Comment (michplunkett) [reply new_arrivals_chi/app/database.py:114]: For your created by fields:
```python
created_at = Column(
        DateTime(timezone=True),
        nullable=False,
        server_default=sql_func.now(),
    )
```

File: new_arrivals_chi/app/config.py
Comment (michplunkett) [reply new_arrivals_chi/app/config.py:7]: Is `DEVELOPMENT` the environment flagging value?

File: new_arrivals_chi/app/config.py
Comment (michplunkett) [inline new_arrivals_chi/app/config.py:11]: Those strings feel like they could be constants.

File: new_arrivals_chi/app/database.py
Comment (michplunkett) [reply new_arrivals_chi/app/database.py:183]: If all `ServiceDate`s take place at a specific location, I think it's worth adding. You could potentially make it nullable in case the respective location isn't present yet for the service date.

**Comment by FedericoDM** [comment]
@klinkoberstar @MadeleineKRoberts I think all of the changes have been addressed!

**Comment by michplunkett** [comment]
A means to improve this process in the future would be to simplify the migration process via some from of `make` command or custom script. I often find that if a command requires more than 3-4 distinct parts of it to run, people will have trouble remembering it (that's the case for me at least).

File: data_model.md
Comment (klinkoberstar) [inline data_model.md:1]: this has been moved into 'docs/model'! don't want this to hold up this merge tho we can always fix this later

**Review by klinkoberstar — APPROVED** [review]
awesome work @FedericoDM ! I know you've been working so hard on this

**Comment by FedericoDM** [comment]
:(

**Comment by michplunkett** [comment]
Taking a look at it now.

**Comment by michplunkett** [comment]
Actually, you can do this all yourself, @FedericoDM: https://noplacelikelocalhost.medium.com/testing-crud-operations-with-sqlite-a-time-saving-guide-for-developers-7c74405d63d5#f53b

File: new_arrivals_chi/app/main.py
Comment (michplunkett) [inline new_arrivals_chi/app/main.py:112]: It sets your default `DATABASE_URL` to the in-memory SQLlite if no other environment variable value is present.

File: tests/conftest.py
Comment (michplunkett) [inline tests/conftest.py:11]: These values are being set twice in your code. I would determine how to do this once, as this could create some frustrating debug issues down the line.

File: new_arrivals_chi/migrations/versions/2024-04-25-2341_16cf46fcdfbf_initial_migration.py
Comment (michplunkett) [inline new_arrivals_chi/migrations/versions/2024-04-25-2341_16cf46fcdfbf_initial_migration.py:1]: 🔬  Since you don't have any other migrations in your main branch, for the sake of clarity, I think it make be a good move to delete all the current migrations and create just one. That way the initial state of the DB is clear.

**Comment by michplunkett** [comment]
You should also update the CODEOWNERS file so that data-migrations are tracked to the backend team.

**Comment by aaronhaefner** [comment]
Is there an `init` command that I should run before the initial migration?

```
(new-arrivals-chi-py3.11) bash-3.2$ flask --app app/main.py db migrate -m "Initial migration"
INFO  [alembic.runtime.migration] Context impl SQLiteImpl.
INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
ERROR [flask_migrate] Error: Target database is not up to date.
```

**Comment by FedericoDM** [comment]
> Is there an `init` command that I should run before the initial migration?
> 
> ```
> (new-arrivals-chi-py3.11) bash-3.2$ flask --app app/main.py db migrate -m "Initial migration"
> INFO  [alembic.runtime.migration] Context impl SQLiteImpl.
> INFO  [alembic.runtime.migration] Will assume non-transactional DDL.
> ERROR [flask_migrate] Error: Target database is not up to date.
> ```

Hello @aaronhaefner you don't need to run the migrations :) the changes are already on the database. In case you want to change something to the database, you would do the following:

- Make the changes in database.py
- Run `flask --app app/main.py db migrate -m "<your migration name>"`
- Run `flask --app app/main.py db upgrade`
- Check the python script that was automatically generated (it lives on the `migrations/versions` folder.

However, this is only necessary if you want to make changes to the database. I will share the credentials so you can connect.

**Review by MadeleineKRoberts — APPROVED** [review]
Such great work Fede!! Only thing to add is a class and method docstrings (I will make a ticket for it)

**Pull request merged**
